pkg/testing/define/define_all.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !define && !local
+
+package define
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/version"
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+func defineAction(t *testing.T, req Requirements) *Info {
+	// always validate requirement is valid
+	if err := req.Validate(); err != nil {
+		panic(fmt.Sprintf("test %s has invalid requirements: %s", t.Name(), err))
+	}
```

Comment:
Shouldn't we check if `req.Sudo` is true and `utils.HasRoot()` here?  For example, we could have a test that requires sudo but is executing on a VM.

Reply:
I didn't add it here because I was going to have trust that the test runner would execute the tests correctly. Just like I don't check the runtime. I can add almost all the same logic that is in the local implementation, if you think we should guard here as well.

Reply:
> I can add almost all the same logic that is in the local implementation

Thinking about this point, I wonder if we even need both `define_local.go` and `define_all.go`?  After all, conceptually, all tests are running locally relative to where `go test` will be executed.  The difference is more on whether we run `go test` locally (developer machine) or we provision a VM, copy over the files, and then run `go test` on the VM.

Reply:
Well we would still need to know the difference between the two so locally we don't run the tests where `local: false` which is the default. I expect more test to be `local: false` then `local: true`.

Reply:
I will try to merge them the best I can.